lib/elixir/src/elixir_clauses.erl

```diff
@@ -282,19 +282,20 @@ expand_with_export(Meta, Kind, _Fun, {Key, _}, _Acc, E) ->
 
 %% Expands all -> pairs in a given key but do not keep the overall vars.
 expand_without_export(Meta, Kind, Fun, Clauses, E) ->
-  NewKind = case lists:keyfind(origin, 1, Meta) of
-              {origin, K} -> K;
-              _ -> Kind
-            end,
-  do_expand_without_export(Meta, NewKind, Fun, Clauses, E).
+  NewKind =
+    case lists:keyfind(origin, 1, Meta) of
+      {origin, Origin} -> Origin;
+      _ -> Kind
+    end,
+  expand_without_export_origin(Meta, NewKind, Fun, Clauses, E).
 
-do_expand_without_export(Meta, Kind, Fun, {Key, Clauses}, E) when is_list(Clauses) ->
+expand_without_export_origin(Meta, Kind, Fun, {Key, Clauses}, E) when is_list(Clauses) ->
```

Comment:
@kelvinst we're trying to get rid of all current occurrences of `do_` prefixing, and not ot introduce new ones. Maybe `expand_without_export_origin` would be better name?

Reply:
Cool! Good to know!